mlir/lib/Dialect/Rock/utility/fusionUtils.cpp

```diff
@@ -0,0 +1,86 @@
+//===- fusionUtils.cpp - Rock utility for fusion -----------------===//
+//
+// Part of the MLIR Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===-----------------------------------------------------===//
+
+#include "mlir/Dialect/Rock/utility/fusionUtils.h"
+#include "mlir/Analysis/BufferDependencyAnalysis.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Rock/IR/Rock.h"
+#include "mlir/Dialect/Rock/Tuning/GridwiseGemmParams.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/Pass/AnalysisManager.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Support/LogicalResult.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/Support/Casting.h"
+
+using namespace mlir;
+using namespace mlir::rock;
+
+struct TestFusibilityPass
+    : public PassWrapper<TestFusibilityPass, OperationPass<func::FuncOp>> {
+  void runOnOperation() override {
+    BufferDependencyAnalysis &analysis =
+        getAnalysis<BufferDependencyAnalysis>();
```

Comment:
Note that `getAnalysis()` implicitly passes in the result of `getOperation()` to the pass constructor.

Reply:
done